Examples/ImageRegistrationMethodBSpline2/ImageRegistrationMethodBSpline2.cxx

```diff
@@ -0,0 +1,139 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// A SimpleITK example demonstrating image registration using the
+// BSplineTransform and the MattesMutualInformation metric.
+
+#include <SimpleITK.h>
+#include <iostream>
+#include <stdlib.h>
+#include <iomanip>
+
+namespace sitk = itk::simple;
+
+class IterationUpdate
+{
+public:
+  IterationUpdate(const sitk::ImageRegistrationMethod & m)
+    : m_Method(m)
+  {}
+
+  void
+  Execute()
+  {
+    std::cout << std::setw(3) << m_Method.GetOptimizerIteration();
+    std::cout << " = " << std::setw(10) << std::setprecision(5) << m_Method.GetMetricValue() << std::endl;
+    std::cout << "\t#: " << m_Method.GetOptimizerPosition().size() << std::endl;
+  }
+
+private:
+  const sitk::ImageRegistrationMethod & m_Method;
+};
+
+class MultiResolutionIterationUpdate
+{
+public:
+  MultiResolutionIterationUpdate(const sitk::ImageRegistrationMethod & m)
+    : m_Method(m)
+  {}
+
+  void
+  Execute()
+  {
+    std::cout << "--------- Resolution Changing ---------" << std::endl;
+  }
+
+private:
+  const sitk::ImageRegistrationMethod & m_Method;
+};
+
+int
+main(int argc, char * argv[])
+{
+
+  if (argc < 4)
+  {
+    std::cerr << "Usage: " << argv[0] << " <fixedImageFile> <movingImageFile> <outputTransformFile>" << std::endl;
+    return 1;
+  }
+
+  sitk::Image fixed = sitk::ReadImage(argv[1], sitk::sitkFloat32);
+  sitk::Image moving = sitk::ReadImage(argv[2], sitk::sitkFloat32);
+
+  std::vector<unsigned int> transformDomainMeshSize(moving.GetDimension(), 10);
+  sitk::Transform           tx = sitk::BSplineTransformInitializer(fixed, transformDomainMeshSize);
+
+  std::cout << "Initial Parameters:" << std::endl;
+  std::vector<double> initialParams = tx.GetParameters();
+  std::cout << "[";
+  std::cout << std::setprecision(5) << std::fixed << initialParams[0];
+  for (size_t i = 1; i < initialParams.size(); i++)
+  {
+    std::cout << ", " << std::setprecision(5) << std::fixed << initialParams[i];
+  }
+  std::cout << "]" << std::endl;
+
+  sitk::ImageRegistrationMethod R;
+  R.SetMetricAsMattesMutualInformation(50);
+
+  double       learningRate = 5.0;
+  unsigned int numberOfIterations = 100;
+  double       convergenceMinimumValue = 1e-4;
+  unsigned int convergenceWindowSize = 5;
+  R.SetOptimizerAsGradientDescentLineSearch(
+    learningRate, numberOfIterations, convergenceMinimumValue, convergenceWindowSize);
+  R.SetOptimizerScalesFromPhysicalShift();
+  R.SetInitialTransform(tx);
+  R.SetInterpolator(sitk::sitkLinear);
+
+  R.SetShrinkFactorsPerLevel(std::vector<unsigned int>{ 6, 2, 1 });
+  R.SetSmoothingSigmasPerLevel(std::vector<double>{ 6, 2, 1 });
```

Comment:
These values may need a ".0" appended to avoid a conversion warning.

Reply:
Done.